docs/java-rest/high-level/search/search-template.asciidoc

```diff
@@ -0,0 +1,118 @@
+[[java-rest-high-search-template]]
+=== Search Template API
+
+The search template API allows for searches to be executed from a template based
+on the mustache language, and also for previewing rendered templates.
+
+[[java-rest-high-search-template-request]]
+==== Search Template Request
+
+===== Inline Templates
+
+In the most basic form of request, the search template is specified inline:
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests}/SearchDocumentationIT.java[search-template-request-inline]
+--------------------------------------------------
+<1> The search is executed against the `posts` index.
+<2> The template defines the structure of the search source. It is passed
+as a string because mustache templates are not always valid JSON.
+<3> Before running the search, the template is rendered with the provided parameters.
+
+===== Registered Templates
+
+Search templates can be registered in advance through stored scripts API. Note that
+the stored scripts API is not yet available in the high-level REST client, so in this
+example we use the simple REST client.
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests}/SearchDocumentationIT.java[register-script]
+--------------------------------------------------
+
+Instead of providing an inline script, we can refer to this registered template in the request:
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests}/SearchDocumentationIT.java[search-template-request-stored]
+--------------------------------------------------
+
+===== Rendering Templates
+
+Given parameter values, a template can be rendered without executing a search:
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests}/SearchDocumentationIT.java[render-search-template-request]
+--------------------------------------------------
+<1> Setting `simulate` to `true` causes the search template to only be rendered.
+
+Both inline and pre-registered templates can be rendered.
+
+===== Optional Arguments
+
+As in standard search requests, the `explain` and `profile` options are supported:
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests}/SearchDocumentationIT.java[search-template-request-options]
+--------------------------------------------------
+
+===== Additional References
+
+The https://www.elastic.co/guide/en/elasticsearch/reference/current/search-template.html#search-template[Search Template HTTP documentation]
```

Comment:
I wasn't sure how to do a proper `asciidoc` link here, as I don't think these two sets of documentation are built together.

Reply:
`{ref}/search-template.html[Search Template documentation]` should do it. That will also go to the proper branch. I would also remove the `HTTP` part from the description.

Reply:
Thanks, I updated this reference.